rust/tools/authorization-logic/src/ast.rs

```diff
@@ -72,10 +72,26 @@ pub enum AstFact {
     AstCanSayFact { p: AstPrincipal, f: Box<AstFact> },
 }
 
+#[derive(Copy, Debug, Clone, Serialize, Deserialize)]
+pub enum AstBinop {
+    LessThan,
+    GreaterThan,
+    Equals,
+    NotEquals,
+    LessOrEquals,
+    GreaterOrEquals
+}
+
+#[derive(Debug, Clone, Serialize, Deserialize)]
+pub enum AstRValue {
+    FlatFactRValue { flat_fact: AstFlatFact},
+    BinopRValue { lnum: String , binop: AstBinop, rnum: String }
```

Comment:
I think that numbers are different from, and more-special-than, regular `String`s and should be represented as such in the AST. You should have a special `struct` wrapper, something like `NumberLiteral`, and put the representation of the number in there. If we're not doing anything number-y with the `NumberLiteral`, a `String` is fine, but usually it's helpful to parse them and have the numeric interpreation in a first-class `int64` (or whatever is appropriate).

Reply:
These can also be variables not just numeric literals.

Reply:
Doesn't that make it even more important to parse, discover, and add structured types around whatever lies underneath this `BinOpRValue`?

Reply:
A separate AST node for `pred_args` (which can be arguments to either predicates or these comparisons) does seem better. I opened an issue to track this: https://github.com/google-research/raksha/issues/429. Because the point of this rust implementation is to just work briefly for a demo, it's probably more important that this improvement makes its way into the C++ version. The issue can track both implementations.

Reply:
OK, that works for me :+1: